### PR TITLE
[RG-142] Fixed reset compile steps

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
Fixed reset compile steps.

### Which part of the code base require a change

- [x] Library: compiler
- [ ] Plug-in: <Specify the plugin name>
- [ ] Engine
- [ ] Documentation
- [ ] Regression tests
- [ ] Continous Integration (CI) scripts

### Screenshot
image

### CODE REVIEW

File: src/Compiler/TaskManager.h [hunk 1]
```diff
@@ -111,12 +111,12 @@ class TaskManager : public QObject {
...
   void run();
   void reset();
-  void rollBack(Task *t);
+  void cleanDownStreamStatus(Task *t);
 
  private:
   QMap<uint, Task *> m_tasks;
   QVector<Task *> m_runStack;
-  QMap<Task *, QVector<Task *>> m_rollBack;
+  QVector<Task *> m_taskQueue;
```
Comment (ravic-rs) [inline src/Compiler/TaskManager.h:119]: What's the motivation to replace map with a vector ?

File: src/Compiler/TaskManager.cpp [hunk 2]
```diff
@@ -228,10 +221,17 @@ void TaskManager::reset() {
...
 }
 
-void TaskManager::rollBack(Task *t) {
-  const auto &roll_back = m_rollBack.value(t);
-  for (const auto &task : roll_back) {
-    task->setStatus(TaskStatus::None);
+void TaskManager::cleanDownStreamStatus(Task *t) {
+  for (auto it{m_taskQueue.begin()}; it != m_taskQueue.end(); ++it) {
+    if (*it == t) {
+      // In case clean action, clean parent is required.
```
Comment (ravic-rs) [inline src/Compiler/TaskManager.cpp:227]: What's the behavior when we don't have clean action for a task ?

**Review by ravic-rs — COMMENTED** [review]
Two comments to address

File: src/Compiler/TaskManager.cpp (hunk 2 above)
Comment (KochynVolodymyr) [reply src/Compiler/TaskManager.cpp:227]: nothing happened in this case

File: src/Compiler/TaskManager.h (hunk 1 above)
Comment (KochynVolodymyr) [reply src/Compiler/TaskManager.h:119]: I make it more simple and more general. No need a map anymore

**Review by TarasSydor — APPROVED** [review]

**Pull request merged**